packages/ckeditor5-find-and-replace/src/findandreplaceediting.ts

```diff
@@ -190,29 +198,25 @@ export default class FindAndReplaceEditing extends Plugin {
 		const { editor } = this;
 		const { model } = editor;
 
-		const { findCallback, results } = editor.execute( 'find', callbackOrText );
+		const { findCallback } = editor.execute( 'find', callbackOrText );
 
-		this._activeResults = results;
+		this.listenTo( model.document, 'change:data', () => {
+			const { changedSearchResults } = onDocumentChange( this.state!.results, editor, findCallback );
 
-		// @todo: handle this listener, another copy is in findcommand.js file.
-		this.listenTo( model.document, 'change:data', () => onDocumentChange( this._activeResults!, editor, findCallback ) );
+			if ( changedSearchResults.length && !this.state!.highlightedResult ) {
+				this.state!.highlightedResult = changedSearchResults[ 0 ];
+			}
```

Comment:
I think it would be suitable to move these lines to the `onDocumentChange()` since it's a continuation of that logic.

Reply:
Moved to that function